File: src/main/kotlin/nullcheck/NullCheck.java
package nullcheck;

public class NullCheck {

  public boolean startsWithA1(String str) {
    if(str == null) {
      throw new IllegalArgumentException("null");
    }
    return str.startsWith("A");
  }

  public Boolean startsWithA2(String str) {
    if(str == null){
      return null;
    }
    return str.startsWith("A");
  }

  public boolean startsWithA3(String str) {
    if(str == null) {
      return false;
    }
    return str.startsWith("A");
  }
}

File: src/main/kotlin/nullcheck/NullCheck.kt
package nullcheck

fun main(args: Array<String>) {
  val str: String? = "ABC"
  println(startsWithA(str ?: "ABC"))
  println(startsWithA1(str))
  println(startsWithA2(str))
  println(startsWithA3(str))
}

fun startsWithA(str: String) : Boolean {
  return str.startsWith("A")
}

fun startsWithA1(str: String?): Boolean {
  return str?.startsWith("A") ?: throw IllegalArgumentException("null")
}

fun startsWithA2(str: String?): Boolean? {
  return str?.startsWith("A")
}

fun startsWithA3(str: String?): Boolean {
  return str?.startsWith("A") ?: false
}

fun startsWithA4(str: String?): Boolean {
  return str!!.startsWith("A")
}